Develop

rlr2931 opened the pull request:
Primarily linting and installation fix, but also some additions to CVs

rlr2931 committed:
odmx: nwis: linting (lines too long)

rlr2931 committed:
initial commit of generic csv timeseries datasource module

TODO: add better handling of timezones

rlr2931 committed:
timeseries_csv: add column name cleanup function; keep time column ('timestamp') as datetime type

rlr2931 committed:
timeseries_csv: check for datetime column and skip gracefully (with error reporting) if missing

rlr2931 committed:
timeseries_csv: add equipment_path for use in processing

dougvj committed:
Add routine to pull timestamp from existing csv

dougvj committed:
pipeline: fix --skip-data-sources argument

dougvj committed:
pipeline: fix bug where feeder_db handle is overwritten in some cases

dougvj committed:
Add a get lastest timestamp general function for use in datasources

dougvj committed:
campbell: query the timestamp early to avoid costly read to mem

dougvj committed:
campbell: fix rsync target directory

dougvj committed:
timeseries-csv: make non-absolute pathing for equipment work

dougvj committed:
populate: handle update of items with unique constraints

- Check for unique constraints and match object IDs for upsert
- add new environment var to check that no duplicate entries are made
  across multiple invokes (UPDATE_ONLY=1)

dougvj committed:
check: show if all datapoints are NULL

rlr2931 committed:
print a warning but don't except if a datasource has no exposed datastreams

dougvj committed:
ssi-config: add alternate parameter name capability; rename ssi.sql and remove sql.py

dougvj committed:
devops: rename ssi-sql package to more appropriate ssi-db

roelofversteeg committed:
WIP trying to make datapackage generation work

rlr2931 committed:
ssi-db: linting fixes (documented in top level docstring)

rlr2931 committed:
devops: fix linting for ssi.config (unecessary dunder call)

rlr2931 committed:
devops: lint fix for ssi.googleapiwrapper (raising too general exceptions)

dougvj committed:
ert-api: cascade the deletes of quadrupole list

dougvj committed:
common: db: fix where clause with delete action

dougvj committed:
python-db: revert errant cascade directive, must be done in sql

rlr2931 committed:
devops: odmx: incorporate changes from odmx support libraries; use try-except to import either as ssi or as support library

rlr2931 committed:
nwis-geochem: odmx: initial commit of nwis-geochem datasource (previously in odmx repo). Includes:
- datasource python file
- merged and symlinked cvs and mappers from global projects folder
- add nwis_geochem datasource type to example project data_sources.json

rlr2931 committed:
odmx: add extension property for usgs qualifier codes

rlr2931 committed:
nwis-geochem: hard pin dataretrieval version until timestamp bug is resolved

rlr2931 committed:
odmx: add option for column_type in add_column function in timeseries_ingestion.py

rlr2931 committed:
odmx: update pipeline to use updated config library

rlr2931 committed:
lint: linting fixes from odmx-linting branch of odmx repo

rlr2931 committed:
odmx: add pylintrc for odmx

rlr2931 committed:
odmx: add import-error and no-name-in-module to pylintrc since they incorrectly flag support libraries; fix path to ignore owc modules that are broken

rlr2931 committed:
devops: more linting fixes, and allow php linting for paf-user-project-api linting to fail

dougvj committed:
odmx: make sure build resolves symlinks from ssi/odmx-support merge

rlr2931 committed:
fixup! Revert "devops: fix linting for ssi.config (unecessary dunder call)" and add pylint disable

rlr2931 committed:
odmx: reset non-global feeder db before trying to wipe it

rlr2931 committed:
odmx: don't duplicate propagation of from_scratch inside the run_pipeline function

rlr2931 committed:
odmx: add print statements to help with troubleshooting plm issues

rlr2931 committed:
odmx: fix m1_10 calculated datastream timestamps (convert from s, not default ns); write calculated m6_60 to the database when print line says we do

dougvj committed:
odmx: db: deal with type aliasing breaking beartype config object check

rlr2931 committed:
odmx: move package data folders into package root so they get installed along with the scripts

rlr2931 committed:
fixup! odmx: move package data folders into package root so they get installed along with the scripts and adjust paths in scripts accordingly

rlr2931 committed:
odmx: lbnlsfa: fix feeder name for m6_60 calculated channel; use vprint instead of print more often

rlr2931 committed:
odmx: lbnlsfa: consolidate plm calculation functions

rlr2931 committed:
odmx: remove wipe_feeder option (no longer needed)

roelofversteeg committed:
WIP nwis ingestion

rlr2931 committed:
ssi-db: don't take type(Config) for beartype checking, Config is already a type

rlr2931 committed:
fixup! odmx: pylint disable import checking for ssi.db and odmx.support.db (not present in odmx and python-ssi-db containers, respectively)

dougvj committed:
devops: add non-docker-deps to yaml which apply to caching/local builds

dougvj committed:
odmx: add ssi-python-db library as dependency

rlr2931 committed:
odmx: remove unused sampling features from example project

dougvj committed:
WIP: refactor local backend definition and clean up images

dougvj committed:
webcam: rewrite to conform to latest IOT api

dougvj committed:
python-ssi-db: update db model generator with base class and minor improvements

dougvj committed:
odmx: update data model with new generator

dougvj committed:
odmx: add simple rest api client

dougvj committed:
ssi-python-db: add limit and offset parameters to query helpers and generators

dougvj committed:
odmx: generate new data model with limit and offset parameters

dougvj committed:
odmx: update rest api to support limit and offset, fix minor bugs and linting

dougvj committed:
odmx: add new rest client module for simpler REST queries

dougvj committed:
odmx: rest-api: fix date time parsing, url decoding, and add open/close interval parameter

dougvj committed:
odmx: rest-api: update REST api to pass configuration parameters to subprocesses

dougvj committed:
odmx: add remote sync module which syncs from a remote REST api to local db

rlr2931 committed:
odmx: check for new data before trying to materialize calculated plm datastream; add print statement indicating which datasource is starting processing

roelofversteeg committed:
WIP Adding NWIS geochemistry

roelofversteeg committed:
Code which correctly parses and ingests the qualifiers associated with the USGS geochem data

roelofversteeg committed:
Updates to ingestion which now writes a marginal qa flag for usgs values which have annotated values so we can exclude them easily

dougvj committed:
odmx: remote_sync: improve project header management

dougvj committed:
devops: continued test format conversion

rlr2931 committed:
odmx: move nwis mappers to global mapper json, use this instead of two different hard-coded mappers in code

rlr2931 committed:
odmx: fix missing fstring in logging of timeseries_processing create_view

rlr2931 committed:
odmx: nwis: add 'discontinued' variables to skip list

rlr2931 committed:
odmx: nwis: handle un-mapped variables the same as nwis_geochem (log and report but do not break pipeline)

rlr2931 committed:
odmx: nwis_geochem: only use nwis_geochem feeder tables for geochem processing

rlr2931 committed:
odmx: nwis_geochem: use usgs remark and value qualifier codes from jsons, not hardcoded lists (which were missing some values)

dougvj commented on setup.py:
```diff
@@ -17,7 +17,9 @@
-    packages=['odmx', 'odmx.datasources'],
+    packages=['odmx', 'odmx.datasources', 'odmx.support'] + \
+        # SSI Specific
+        ['odmx.datasources.ssi', 'odmx.ssi'],
```
I believe this needs to be removed

dougvj committed:
devops: bump pandas version to solve ODMX nwis issues

dougvj committed:
python: db: do not pass sql DBAPI object into pandas due to breakage from update

rlr2931 committed:
odmx: update docstrings in timeseries_csv with more detail

rlr2931 committed:
odmx: move mappers into odmx repo data so they aren't dependent on workdir

rlr2931 committed:
odmx: move decagon mappers into standalone jsons (not hardcoded in datasource script)

dougvj committed:
python-db-lib: fix typo preventing the delete function from working

rlr2931 committed:
odmx: replace ssi.general with odmx.file_utils (which overlaps with ssi.general but has only the functions used in the odmx pipeline

rlr2931 committed:
odmx: snotel: move mappers into standalone json files

rlr2931 committed:
fixup! odmx: replace ssi.general with odmx.file_utils (which overlaps with ssi.general but has only the functions used in the odmx pipeline; use importlib.util.find_spec to find mappers and other package data

rlr2931 committed:
fixup! odmx: replace ssi.general with odmx.file_utils (which overlaps with ssi.general but has only the functions used in the odmx pipeline; use resource_filename.pkg_resources to find mappers and other package data

rlr2931 committed:
odmx: update to latest usgs dataretrieval tag for compatibility with newer pandas (bug that cused pin to 1.0.3 has been fixed)

rlr2931 committed:
odmx: decagon: remove duplicate entries in decagon variables mapper

rlr2931 committed:
odmx: remove pointless get_mapping function, get position_start for nwis from data instead of hardcoded value copied from decagon

rlr2931 committed:
wip: odmx: add function to check for existing equipment and data to equipment maps because we should do this for all data source types not just decagon; move check_diff to write_equipment_jsons, remove no-longer-used get_mapping function

rlr2931 committed:
odmx: convert timeseries_csv datasource to generic_timeseries and use data_source_config.json that will be paired with data file

rlr2931 committed:
odmx: hydrovu: move unit and variable mappings to standalone json, find mappers using importlib.util find_spec

rlr2931 committed:
odmx: move general data_to_equipment_map starter to write_equipment_jsons script

rlr2931 committed:
odmx: add unit arg to get_csv_last_timestamp for datasources with timestamps stored not in ns (decagon)

rlr2931 committed:
odmx: update docstrings and lint file_utils

rlr2931 committed:
odmx: don't hold mapper files open while checking diffs, if file doesn't exist write it

rlr2931 committed:
ssi: odmx: decagon: add Barometer to equipment mapper

rlr2931 committed:
odmx: decagon: update kilopascals in units mapper to match cv term

rlr2931 committed:
odmx: add helper function for creating data_source_config.json from csv column headers

rlr2931 committed:
odmx: timeseries_ingestion: change applymap to map per FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead

rlr2931 committed:
odmx: homogenize writing of equipment.json and data_to_equipment_map.json across data source types, remove duplicate clean_name functions (now in support/file_utils.py)

rlr2931 committed:
odmx: major overhaul of timeseries_csv
- can use either csv or excel files
- add use of data_source_config.json (stored alongside datafile) for mapping columns/ creating equipment jsons
- remove duplicate functions (now in file_utils)
- clean up docstrings

rlr2931 committed:
wip: odmx: linting

rlr2931 committed:
odmx: split out core geochem ingestion functions so nwis_geochem can use them in the public repo

dougvj committed:
odmx: snotel: be flexible about whether a timestamp has a time element

rlr2931 committed:
odmx: add additional special characters to clean_names, add function for expanding colun names specified using *

rlr2931 committed:
wip: odmx: hydrovu: update to use new equipment json writing approach

rlr2931 committed:
odmx: update generic_timeseries datasource with cleaner pathing and mappings

rlr2931 committed:
odmx: nwis: use correct index for finding first timestamp in nwis data

rlr2931 committed:
odmx: hydrovu: update writing of equipment jsons to match how it's done in other datasources

rlr2931 committed:
odmx: file_utils: fix docstring typo

rlr2931 committed:
odmx: generic_timeseries: handle column name mappings more cleanly, improvve inline commenting

rlr2931 committed:
odmx: move routine for writing generic equipment jsons from data_source_config to write_equipment_jsons.py

rlr2931 committed:
odmx: add update_equipment_jsons to example project data_sources.json

rlr2931 committed:
odmx: snotel: add catch for last_server_time being read as a string; clean up variables that are multiply defined

dougvj committed:
odmx: fix non-numeric timestamp handling for csv last timestamp lookup

dougvj committed:
odmx: use ODMX_ prefix for env var and clear empty rows from cambpell

dougvj committed:
devops: update pip for beartype, h5py

dougvj committed:
ssi-support: db: add support for returning row count from queries

dougvj committed:
odmx: update datamodel with new generator which supports row counts

dougvj committed:
odmx: update rest API to support required user ID and row count in json list

dougvj committed:
Remove unecessary README in subdir

dougvj committed:
snowtel: handle ISOdatetimes with both T and space separators

dougvj committed:
readme: add blurb about limit and offset

dougvj committed:
Update README with information on pagination and minor enhancements

dougvj committed:
example: remove extraneous field for nwis_geochem

dougvj committed:
Bump version for updated release

dougvj committed:
Remove build.yml used with internal SSI tooling
